store/src/java/com/zimbra/cs/filter/RuleManager.java

```diff
@@ -414,9 +415,37 @@ private static void setXMLRules(Account account,List<FilterRule> rules, String s
                     // Don't apply user filters to spam by default
                     applyRules = false;
                 }
-
                 if (applyRules) {
-                    SIEVE_FACTORY.evaluate(mailAdapter, node);
+                    if (filter.equals(FILTER_RULES_CACHE_KEY)) {
+                        mailAdapter.setUserScriptExecuting(true);
+                    }
+                    try {
+                        SIEVE_FACTORY.evaluate(mailAdapter, node);
+                    } catch (SieveException e) {
+                        if (editHeaderUserScriptError.equals(e.getMessage())) {
```

Comment:
lines 425 to 448 can be moved to a common method as same code is repeated at line 497 to 526.